src/v/kafka/server/group.cc

```diff
@@ -1870,6 +1875,14 @@ group::prepare_tx(cluster::prepare_group_tx_request r) {
         ptx.offsets[tp] = md;
     }
     _prepared_txs[r.pid] = ptx;
+
+    auto exp_it = _expiration_info.find(r.pid);
+    vassert(
+      exp_it != _expiration_info.end(),
+      "pid({}) should be inside _expiration_info",
+      r.pid); // Maybe return another error?
+    exp_it->second.update_last_update_time();
```

Comment:
A question here..  what does it mean to expire a prepared txn? This is a PONR (point of no return) right? If the both rm_stm#prepare and group#prepare succeed, we ack the client that the transaction is success, we can't expire after that, right?

Reply:
when a tx in a prepared state expires it communicates with the coordinator to find out its true state.